cmd/kubeadm/app/cmd/reset.go

```diff
@@ -68,15 +69,21 @@ func NewCmdReset(in io.Reader, out io.Writer) *cobra.Command {
 				kubeadmutil.CheckErr(err)
 			}
 
+			cfg, err := configutil.FetchInitConfigurationFromCluster(client, os.Stdout, "reset", false)
+			if err != nil {
+				cfg = nil
+				klog.Warningf("[reset] Unable to fetch the kubeadm-config ConfigMap from cluster: %v", err)
+			}
+
 			if criSocketPath == "" {
-				criSocketPath, err = resetDetectCRISocket(client)
+				criSocketPath, err = resetDetectCRISocket(cfg)
 				kubeadmutil.CheckErr(err)
 				klog.V(1).Infof("[reset] detected and using CRI socket: %s", criSocketPath)
 			}
 
 			r, err := NewReset(in, ignorePreflightErrorsSet, forceReset, certsDir, criSocketPath)
 			kubeadmutil.CheckErr(err)
-			kubeadmutil.CheckErr(r.Run(out, client))
+			kubeadmutil.CheckErr(r.Run(out, client, cfg))
```

Comment:
i don't think we should call `Run` if `cfg` is `nil`

Reply:
We not call `Run` if cfg is nil, reset will be aborted. 
I see that in the `Run` workflow, the next processes will continue, even though have the mistakes of the previous step.

Reply:
yes, i guess that's true.
i guess we can pass `nil` here but inside Run we need to not use `cfg.SomeField` because it will panic.

Reply:
I check the currently code, I find `resetDetectCRISocket` , `getEtcdDataDir` and `RemoveStackedEtcdMemberFromCluster` using `cfg`, and we should check `cfg` is not nil in these place if we pass `nil` here.
Should i check `cfg` is nil before invoke `Run`, or check it inside Run? Which one do you prefer?

Reply:
so we should run all the code that doesn't depend on `cfg`, but if a line is calling `cfg.SomeField` we should always check if `cfg` is `nil` before that.

Reply:
yes, you said right.

I update the code in my local development environment, below is code:

```go
cfg, err := configutil.FetchInitConfigurationFromCluster(client, os.Stdout, "reset", false)
if err != nil {
	kubeadmutil.CheckErr(errors.Errorf("[reset] Unable to fetch the kubeadm-config ConfigMap from cluster: %v\n", err))
}
```

If we check `cfg` is `nil` , reset will be aborted. If user wants to continue cleanup, should we tell them how to do it?

Reply:
replied here: https://github.com/kubernetes/kubernetes/pull/74410/files#r260095993